inbm/dispatcher-agent/dispatcher/source/linux_gpg_key.py

```diff
@@ -61,4 +62,4 @@ def add_gpg_key(remote_key_path: str, key_store_name: str) -> None:
         raise SourceError(f"Error getting GPG key from remote source: {e}")
 
     except subprocess.CalledProcessError as e:
-        raise SourceError(f"Error running gpg command to dearmor key: {e}")
+        raise SourceError(f"Error running GPG command to dearmor key: {e}")
```

Comment:
**Best Practice**
**Suggestion:** Use a context manager to ensure the file is closed properly after writing the key.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    with open(key_path, 'w') as key_file:
        subprocess.run(
            ["/usr/bin/gpg", "--dearmor"],
            input=decoded_ascii,
            stdout=key_file,
            check=True,
            text=True,
            shell=False,
        )
```